kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/CreateExternalOrderCall.java

```diff
@@ -0,0 +1,124 @@
+package com.kin.ecosystem.data.order;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+import com.kin.ecosystem.Callback;
+import com.kin.ecosystem.base.Observer;
+import com.kin.ecosystem.data.blockchain.IBlockchainSource;
+import com.kin.ecosystem.data.model.Payment;
+import com.kin.ecosystem.network.ApiException;
+import com.kin.ecosystem.network.model.JWTBodyConfirmPaymentResult;
+import com.kin.ecosystem.network.model.OpenOrder;
+import com.kin.ecosystem.network.model.Order;
+import com.kin.ecosystem.util.ExecutorsUtil.MainThreadExecutor;
+import java.math.BigDecimal;
+
+class CreateExternalOrderCall extends Thread {
+
+    private static final String TAG = CreateExternalOrderCall.class.getSimpleName();
+
+    private OrderDataSource.Remote remote;
+    private IBlockchainSource blockchainSource;
+    private String orderJwt;
+    private ExternalOrderCallbacks externalOrderCallbacks;
+
+    private OpenOrder openOrder;
+    private MainThreadExecutor mainThreadExecutor = new MainThreadExecutor();
+
+    public CreateExternalOrderCall(@NonNull OrderDataSource.Remote remote, @NonNull IBlockchainSource blockchainSource,
+        @NonNull String orderJwt,
+        @NonNull ExternalOrderCallbacks externalOrderCallbacks) {
+        this.remote = remote;
+        this.blockchainSource = blockchainSource;
+        this.orderJwt = orderJwt;
+        this.externalOrderCallbacks = externalOrderCallbacks;
+    }
+
+    @Override
+    public void run() {
+        try {
+            // Create external order
+            openOrder = remote.createExternalOrderSync(orderJwt);
+        } catch (final ApiException e) {
```

Comment:
if sever blocks external order will it thrown ApiException? what happen if server determined that this exact external order is either open, submitted  or completed? if completed getOrder should be called and return JWT externalOrderCallbacks  .  if already submitted need to wait to  for blockchain conformation and than getOrder with JWT ....

Reply:
we can open a task for error handling, right now it will pass the error to the client.